src/rt/region/region_arena.h

```diff
@@ -578,6 +580,21 @@ namespace verona::rt
       dealloc(alloc);
     }
 
+    /*
+     * Unlike traced regions, objects with a destructor but no finaliser are
+     * placed with the objects that do have a finaliser. This avoids having to
+     * walk all the objects on deallocation to find the ones with destructors.
+     */
+    static bool needs_finaliser_area(const Descriptor* desc)
```

Comment:
I don't think we ever agreed on what to call the finaliser and non-finaliser... areas... of an arena. My only nitpick with `area` is that it looks too much like `arena`. (`non-finaliser` is also a bit of a mouthful to say, and I guess it's no longer true, now that objects without finalisers but with destructors go into the other end.)

Reply:
I think we had discusses the C++ non-trivially destructable.  But that is even more of a mouthful.

Rust uses the "Drop" term, so we could perhaps TrivialDrop and NonTrivialDrop.